src/main/java/com/wanderersoftherift/wotr/block/DittoBlock.java

```diff
@@ -0,0 +1,169 @@
+package com.wanderersoftherift.wotr.block;
+
+import com.mojang.serialization.MapCodec;
+import com.wanderersoftherift.wotr.block.blockentity.DittoBlockEntity;
+import com.wanderersoftherift.wotr.init.ModBlocks;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.particles.ParticleTypes;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.stats.Stats;
+import net.minecraft.world.Containers;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.BlockItem;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.BaseEntityBlock;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.RenderShape;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
+import net.minecraft.world.level.gameevent.GameEvent;
+import net.minecraft.world.phys.BlockHitResult;
+import net.neoforged.neoforge.registries.DeferredBlock;
+
+import javax.annotation.Nullable;
+
+public class DittoBlock extends BaseEntityBlock {
+	public static final MapCodec<DittoBlock> CODEC = simpleCodec(DittoBlock::new);
+	public static final BooleanProperty HAS_ITEM = BooleanProperty.create("has_item");
+
+
+	public MapCodec<DittoBlock> codec() {
+		return CODEC;
+	}
+
+	public DittoBlock(BlockBehaviour.Properties p_273064_) {
+		super(p_273064_);
+		this.registerDefaultState(this.defaultBlockState().setValue(HAS_ITEM, false));
+	}
+
+	@Override
+	protected RenderShape getRenderShape(BlockState state) {
+		return RenderShape.INVISIBLE;
+	}
+
+	protected InteractionResult useWithoutItem(BlockState state, Level level, BlockPos pos, Player player, BlockHitResult hitResult) {
+		BlockEntity blockEntity = level.getBlockEntity(pos);
+		if (!(blockEntity instanceof DittoBlockEntity dittoBlockEntity))
+			return InteractionResult.PASS;
+
+		if (level.isClientSide)
+			return InteractionResult.SUCCESS;
+
+		ItemStack itemstack1 = dittoBlockEntity.getTheItem();
+		if (!itemstack1.isEmpty()) {
+			ItemStack itemstack = getBlock().toStack();
+			if (dittoBlockEntity.isEmpty()) {
+				return InteractionResult.PASS;
+			} else {
+				if (itemstack1.getItem() != getBlock().asItem())
+					Containers.dropContents(level, pos, dittoBlockEntity);
+				else
+					return InteractionResult.PASS;
+				dittoBlockEntity.setTheItem(itemstack);
+			}
+
+			level.playSound((Player) null, pos, SoundEvents.DECORATED_POT_INSERT, SoundSource.BLOCKS, 1.0F, 0.5F);
+			if (level instanceof ServerLevel) {
```

Comment:
Could use `if (level instanceof ServerLevel serverLevel)` here and remove the next line.